application/src/main/java/com/sanction/thunder/resources/UserResource.java

```diff
@@ -114,51 +124,43 @@ public Response updateUser(@Auth Key key, PilotUser user) {
   /**
    * Retrieves a PilotUser from the database.
    *
-   * @param username The username of the user to retrieve.
-   * @return The user that was found in the database.
+   * @return The pilotUser that was found in the database.
    */
   @GET
-  public Response getUser(@Auth Key key, @QueryParam("username") String username) {
+  public Response getUser(@Auth User user) {
     getRequests.mark();
 
-    if (username == null || username.equals("")) {
-      return Response.status(Response.Status.BAD_REQUEST)
-          .entity("username query parameter is required to get a user.").build();
-    }
+    String username = user.getName();
```

Comment:
You'll get a 401 if you don't provide credential at all and a 401 if the user doesn't exist